pkg/lwip/contrib/sys_arch.c

```diff
@@ -178,19 +178,46 @@ u32_t sys_arch_mbox_tryfetch(sys_mbox_t *mbox, void **msg)
     }
 }
 
+/**
+ * @brief   parameters to _lwip_thread_wrapper
+ */
+typedef struct {
+    mutex_t sync;
+    lwip_thread_fn thread;
+    void *arg;
+} _lwip_thread_params_t;
+
+static void *_lwip_thread_wrapper(void *params_ptr)
+{
+    _lwip_thread_params_t *params = params_ptr;
+    lwip_thread_fn thread = params->thread;
+    void *arg = params->arg;
+    mutex_unlock(&params->sync);
+    thread(arg);
+    /* TODO: free stack? */
```

Comment:
@kaspar030 any idea how we might do that? It's allocated [here](https://github.com/RIOT-OS/RIOT/pull/9588/files#diff-a8e8f0f49dbd0ad3cf6764a585ec54acR205)

Reply:
If we were sure that the thread is not going to be forcefully cancelled, it could be done in the wrapper, before returning.

Reply:
Depending on the architecture, the stack might still be used to write the return value, so this sounds like a bad idea ;-).

Reply:
Well, at the moment non of the threads started by lwIP terminate, so this is a more theoretical issue in case lwIP ever decides to have something that actually finishes ;-).